packages/flutter/test/widgets/box_decoration_test.dart

```diff
@@ -342,4 +343,218 @@ Future<void> main() async {
+
+    await tester.tapAt(borderTopTangent);
+    expect(itemsTapped, <int>[1],reason: 'border Top not tapped');
+
+    await tester.tapAt(borderLeftTangent);
+    expect(itemsTapped, <int>[1,1],reason: 'border Left not tapped');
+
+    await tester.tapAt(fartherBorderRadiusPoint);
+    expect(itemsTapped, <int>[1,1,1],reason: 'border center not tapped');
+
+    await tester.tap(find.byKey(key));
+    expect(itemsTapped, <int>[1,1,1,1]);
+  });
+
+  testWidgets('BoxDecoration rounded angles other corner works', (WidgetTester tester) async {
+    const double height = 50.0;
+    const double width = 50.0;
+    const double radius = 20;
+
+    List<int> itemsTapped;
+    const Key key = Key('Container with BoxDecoration');
+    Widget buildFrame(Border border) {
+      itemsTapped = <int>[];
+      return Align(
+          alignment: Alignment.topLeft,
+          child:GestureDetector(
+            behavior: HitTestBehavior.deferToChild,
+            child: Container(
+                key: key,
+                width: width,
+                height: height,
+                decoration: BoxDecoration(border: border, shape: BoxShape.rectangle,borderRadius: BorderRadius.circular(radius))
+            ),
+            onTap: () {
+              itemsTapped.add(1);
+            },
+          )
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(Border.all()));
+
+    expect(itemsTapped, isEmpty);
+
+    await tester.tap(find.byKey(key));
+    expect(itemsTapped, <int>[1]);
+
+    // x, y
+    const Offset topRightOutside = Offset(width, 0.0);
+    const Offset topRightInside = Offset(width-radius, radius);
+    const Offset bottomRightOutside = Offset(width, height);
+    const Offset bottomRightInside = Offset(width-radius, height-radius);
+    const Offset bottomLeftOutside = Offset(0, height);
+    const Offset bottomLeftInside = Offset(radius, height-radius);
+    const Offset topLeftOutside = Offset(0, 0);
+    const Offset topLeftInside = Offset(radius, radius);
+
+    await tester.tapAt(topRightInside);
+    expect(itemsTapped, <int>[1,1],reason: 'top right not tapped');
+
+    await tester.tapAt(topRightOutside);
+    expect(itemsTapped, <int>[1,1],reason: 'top right tapped');
+
+    await tester.tapAt(bottomRightInside);
+    expect(itemsTapped, <int>[1,1,1],reason: 'bottom right not tapped');
+
+    await tester.tapAt(bottomRightOutside);
+    expect(itemsTapped, <int>[1,1,1],reason: 'bottom right tapped');
+
+    await tester.tapAt(bottomLeftInside);
+    expect(itemsTapped, <int>[1,1,1,1],reason: 'bottom left not tapped');
+
+    await tester.tapAt(bottomLeftOutside);
+    expect(itemsTapped, <int>[1,1,1,1],reason: 'bottom left tapped');
+
+    await tester.tapAt(topLeftInside);
+    expect(itemsTapped, <int>[1,1,1,1,1],reason: 'top left not tapped');
+
+    await tester.tapAt(topLeftOutside);
+    expect(itemsTapped, <int>[1,1,1,1,1],reason: 'top left tapped');
+  });
+
+}
```

Comment:
Please remove the extra blank line and the new line at EOF - then we should be good :)

Reply:
done, you should think something for add an automatic formatting to the repository..

Reply:
I know it's a bit of a pain, sorry about that.  You can read more about the rationale at https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#in-defense-of-the-extra-work-that-hand-formatting-entails